Login form and view

subnomo opened the pull request:
Added the login view and a basic template.

subnomo committed:
Add login view

refs #3

subnomo committed:
Add login form

refs #3

pop commented on app/views.py:
```diff
@@ -0,0 +1,40 @@
+        password = request.form['password']
+
+        # Login to TimeSync
+        ts = pymesync.TimeSync(baseurl="http://timesync-staging.osuosl.org/v1")
```
this url should be in conf.py

pop commented:
Looks good to me. A few pieces of feedback (unrelated to the files changed otherwise I would do them inline):
- `README.md` should include a line about 'how to run the app', as well as any setup to do before running in production.
- `conf.py` should really be `conf.py.dist` and the `README.md` will explain what to change before putting the app in production.
- This PR should be apart of the previous one. It's no big deal now, but we try to avoid merging failing tests into develop. It might be different for this project, but I don't remember hearing about us trying that.

The actual content of this PR looks good, there are just a few things to change (listed above).

subnomo committed:
Create .dist for configuration

Also, get TimeSync baseurl from config.

subnomo committed:
Remove THREADS_PER_PAGE config

This config parameter is used in many example projects, but Flask doesn't actually use it.

subnomo committed:
Remove CSRF related options from config and README

CSRF is enabled by default in Flask-WTF so we don't need to specifiy these anymore.

subnomo committed:
Update README with run instructions

subnomo commented:
Some changes coming, so hold off on review.

subnomo committed:
Use Flask-WTF for forms

refs #3

subnomo commented:
Ready for review.

Kennric commented:
Looks pretty good so far. Please  create issues for your TODOs. +1

pop commented:
This is looking pretty good. I would feel a bit better if travis was running our tests instead of just linting though.

matthewrsj commented:
Tests pass and this is looking good, but :-1: until you run our tests with travis.

matthewrsj commented on README.md:
```diff
@@ -9,6 +9,14 @@ $ source venv/bin/activate
+2. Change the `SECRET_KEY` to one of your own. Make it unique.
+4. Change `TIMESYNC_URL` to the URL of the TimeSync instance to
+communicate with.
+5. If you want to run it in debug mode, make sure you set `DEBUG` to `True`.
```
You should also add the command to run the app:

```
python run.py
```

and explain that it is now running on `http://localhost:5000` and the login page can be accessed at `/login`.

matthewrsj commented on app/views.py:
```diff
@@ -0,0 +1,44 @@
+            return 'There was an error.', 500
+
+        session['username'] = username
+        session['ts'] = token
```
Is it going to be confusing having `ts` be a `pymesync.TimeSync` instance but `session['ts']` a JSON object containing `{'token': 'thereallylongtoken'}`?

You may also want to keep in mind that `session['ts']` does not just contain a token, but as explained above, a JSON object _containing_ a token. You will have to keep this in mind for re-auth in other views when you have to pass **just** a token to initiate pymesync.

The alternative to that is to set `session['ts']` (or `session['token']`) to `token['token']` so you don't have to worry about it later. You already verified there is no `error` in `token` so you shouldn't see any key errors. If you want to be even more explicit though you could do something like:

```
if 'token' in token:
    session['token'] = token['token']
else:
    print "unexpected response " + token
```

subnomo committed:
Use bare token

subnomo committed:
Update README with more run instructions

subnomo committed:
Add testing to travis

matthewrsj commented on .travis.yml:
```diff
@@ -0,0 +1,15 @@
+install:
+    - pip install -r requirements.txt
+
+script:
```
You are going to want to add something like this here (before this line):

```
before_script:
  - mv config.py.dist config.py
```

and additionally take the steps to set up the test environment as described in the README.

subnomo committed:
Use pymesync test mode

subnomo committed:
Flake config.py.dist as well

subnomo committed:
Have travis move config.py.dist to config.py

Rather than falling back to config.py.dist in the main application

subnomo committed:
Flake config.py rather than config.py.dist

subnomo commented:
@ElijahCaine @Kennric Review?

matthewrsj commented:
@subnomo you'll need to rebase this. ping me if you need help.

![](http://i.imgur.com/MXjhmsc.png)

pop commented:
Tests++ are always good, the code looks good, everything looks good to me. +1 :+1:

subnomo committed:
Add testing config value for clarity

matthewrsj commented:
This is understandable, tests pass. I think this is ready to merge. +1
